### PR TITLE
Button groups

### PR DESCRIPTION
**Pull request by paulkahle**
Refactored toggle buttons and groups
1.  Added toggle functionality to our standard buttons.  "toggleable" = true enables toggle, "toggled" sets toggle state.
2.  Created a button-group component for grouping buttons (toggleable or otherwise)
3.  If a button group is "selectable" it will toggle the selections like a group of radio buttons.  If it is also "multiple" it will support multiple selections like a group of checkboxes.
4. Moved DataDirective out of toggle-group

I left the old toggle button/group code in for now, but these components should replace those.

### CODE REVIEW

File: src/lib/data-attribute/data-attribute.directive.ts [hunk 1]
```diff
@@ -5,7 +5,7 @@ import { Directive, Input, OnChanges, ElementRef } from '@angular/core';
 @Directive({
   selector: '[supreData]'
 })
-export class DataDirective implements OnChanges {
+export class DataAttributeDirective implements OnChanges {
```
Comment (bcandullo) [inline src/lib/data-attribute/data-attribute.directive.ts]: Is this deprecated now? Seems like you can do all of this with [attr] bindings.

Comment (paulkahle) [reply src/lib/data-attribute/data-attribute.directive.ts]: Is there a way to do attr binding dynamically? I.e. bind attributes without knowing what those attributes are?

**Comment by paulkahle** [comment]
Yeah, I'm going to revisit the disabled style.  I don't really like it right now.

File: src/lib/data-attribute/data-attribute.directive.ts (hunk 1 above)
Comment (bcandullo) [reply src/lib/data-attribute/data-attribute.directive.ts]: Yea that's a little tricky, probably best to keep it in a directive.

**Pull request merged**